doc/xversionmessage.md

```diff
@@ -0,0 +1,250 @@
+XVersionMessage: BCH node extended version and configuration fields
+===================================================================
+
+DRAFT specification
+Version: 0.0.3
+
+Overview
+--------
+
+Using the `version` message in the Bitcoin protocol (for more details,
+see https://en.bitcoin.it/wiki/Protocol_documentation#version), peers
+announce their capabilities and preferences to each other. The
+`version` message is limited in scope to a fixed set of well-known
+fields.
+
+It is deemed desirable to be able to announce further configuration
+and version information to peers. For example, this can then be used
+to version advanced block transmission protocols like Graphene.
+
+Notably, the `version` message can not be extended as there are
```

Comment:
Network clients that don't yet support this pull request (in other words, all network clients) are not "old".  XT just happens to enforce the current protocol more completely than others in this regard.

Reply:
An upgrade schedule can be introduced similar, but not tied, to the hardfork schedule.  Giving nodes time to update to version that support a new version message should alleviate the issue of flagging illegal version messages.

Reply:
I think this just shuffles the responsibility around rather than eliminating it.  Rather than putting communication pressure on the devs <-> node operator edges, introducing a new message type puts them on the dev <-> dev edges between implementations.  Considering that dev resources are spread thin, I'm not sure this is the best place to put them.

Reply:
Hmm, but how does your approach move comms towards devs<->node-ops? It seems to me that in either case, the dev<->node-op comm is necessary. It seems that we actually avoid dev-dev comms as it can be gradually implemented.